tests/vibe.web.rest.1140/source/app.d

```diff
@@ -0,0 +1,52 @@
+import std.datetime;
+import vibe.d;
+
+interface IOrientDBRoot
+{
+	@property IOrientDBQuery query();
+}
+
+interface IOrientDBQuery
+{
+	@method(HTTPMethod.GET)
+	@path(":db_name/sql/:query/:result_set_size")
+	Json sql(string _db_name, string _query, int _result_set_size);
+}
+
+class OrientDBRoot : IOrientDBRoot {
+	private OrientDBQuery m_query;
+	override @property IOrientDBQuery query() { return m_query; }
+	public this() { this.m_query = new OrientDBQuery(); }
+}
+
+class OrientDBQuery : IOrientDBQuery {
+	override Json sql(string _db_name, string _query, int _result_set_size) {
+		assert(_db_name == Param1, _db_name);
+		assert(_query == Param2, _query);
+		assert(_result_set_size == Param3, to!string(_result_set_size));
+		return Json.emptyObject;
+	}
+}
+
+enum Param1 = "twitter_data";
+enum Param2 = "select DownloadedDateTime from Message";
```

Comment:
Yes, it would be wise.
However,  we need to change internal URL behaviour anyway (withoit changing external behaviour).
Now it provides encoding in output method (`localURI`) and sometimes in input methods (sometimes not).
We should provide encoding in _all_ input method (except new `pathEncoded`, which should only check the correct encoding of input string), and we shouldn't provide any encoding in output methods (like `localURI`). Also would be reasonable to add `invariant` to URI, which checks the correct encoding of `m_pathString` everytime.